nifi-docs/src/main/asciidoc/administration-guide.adoc

```diff
@@ -485,98 +485,137 @@ It is preferable to request upstream/downstream systems to switch to https://cwi
+tasks for managing which nodes are allowed in the cluster and providing the most up-to-date flow to newly joining nodes. When a
+DataFlow Manager manages a dataflow in a cluster, they are able to do so through the User Interface of any node in the cluster. Any
+change made is then replicated to all nodes in the cluster.
+
+*Nodes*: Each cluster is made up of one or more nodes. The nodes do the actual data processing.
+
+*Primary Node*: Every cluster has one Primary Node. On this node, it is possible to run "Isolated Processors" (see below).
+ZooKeeper is used to automatically elect a Primary Node. If that node disconnects from the cluster for any reason, a new
+Primary Node will automatically be elected. Users can determine which node is currently elected as the Primary Node by
+looking at the Cluster Management page of the User Interface.
+
+*Isolated Processors*: In a NiFi cluster, the same dataflow runs on all the nodes. As a result, every component in the flow
+runs on every node. However, there may be cases when the DFM would not want every processor to run on every node. The most
+common case is when using a processor that communicates with an external service using a protocol that does not scale well.
+For example, the GetSFTP processor pulls from a remote directory, and if the GetSFTP Processor runs on every node in the
+cluster tries simultaneously to pull from the same remote directory, there could be race conditions. Therefore, the DFM could
+configure the GetSFTP on the Primary Node to run in isolation, meaning that it only runs on that node. It could pull in data and -
+with the proper dataflow configuration - load-balance it across the rest of the nodes in the cluster. Note that while this
+feature exists, it is also very common to simply use a standalone NiFi instance to pull data and feed it to the cluster.
+It just depends on the resources available and how the Administrator decides to configure the cluster.
+
+*Heartbeats*: The nodes communicate their health and status to the currently elected Cluster Coordinator via "heartbeats",
+which let the Coordinator know they are still connected to the cluster and working properly. By default, the nodes emit
+heartbeats every 5 seconds, and if the Cluster Coordinator does not receive a heartbeat from a node within 40 seconds, it
+disconnects the node due to "lack of heartbeat". (The 5-second setting is configurable in the _nifi.properties_ file.
+See the <<system_properties>> section of this document for more information.) The reason that the Cluster Coordinator
+disconnects the node is because the Coordinator needs to ensure that every node in the cluster is in sync, and if a node
+is not heard from regularly, the Coordinator cannot be sure it is still in sync with the rest of the cluster. If, after
+40 seconds, the node does send a new heartbeat, the Coordinator will automatically reconnect the node to the cluster.
```

Comment:
If the node attempts to reconnect, the Cluster Coordinator still ensures the flows are in sync, correct?